caddyconfig/httpcaddyfile/pkiapp.go

```diff
@@ -16,23 +16,108 @@ package httpcaddyfile
 
 import (
 	"github.com/caddyserver/caddy/v2/caddyconfig"
+	"github.com/caddyserver/caddy/v2/caddyconfig/caddyfile"
 	"github.com/caddyserver/caddy/v2/modules/caddypki"
 )
 
+func init() {
+	RegisterGlobalOption("pki", parsePKIApp)
+}
+
+// parsePKIApp parses the global log option. Syntax:
+//
+//     pki {
+//         ca [<id>] {
+//             name                     <name>
+//             root_common_name         <name>
+//             intermediate_common_name <name>
```

Comment:
Just a thought... I wonder if it would be worth renaming these to use `_cn` intead of `_common_name` given that CN is the very common abbreviation for CommonName.

Reply:
Hmm, yeah why not. Shortening is probably a good idea. We don't have people write out "certificate" (cert) or "transport_layer_security" 😅

Reply:
Heh. I still regret not using "re" instead of "regexp" in those places in the Caddyfile.